Cleaned up some code

Saiyato opened the pull request:
And updated uninstall script of kodi

fixes #116

volumio commented:
@Saiyato published, congrats! 

Two things:
- I've added missing UIConfig.json, please pull my code to yours to avoid confilicts
- The plugin uninstall cannot remove  some kodi related stuff, make sure they're owned by volumio
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/temp/temp': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/temp/archive_cache': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/temp/kodi.log': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/temp/kodi.old.log': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/Configuration': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/userdata/profiles.xml': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/userdata/Database/Addons27.db': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/userdata/Database/CDDB': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/userdata/addon_data': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/userdata/library': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/userdata/Thumbnails/c': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/userdata/Thumbnails/b': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/userdata/Thumbnails/a': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/userdata/Thumbnails/4': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/userdata/Thumbnails/e': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/userdata/Thumbnails/2': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/userdata/Thumbnails/d': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/userdata/Thumbnails/3': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/userdata/Thumbnails/7': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/userdata/Thumbnails/9': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/userdata/Thumbnails/0': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/userdata/Thumbnails/1': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/userdata/Thumbnails/5': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/userdata/Thumbnails/Video/Bookmarks': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/userdata/Thumbnails/f': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/userdata/Thumbnails/8': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/userdata/Thumbnails/6': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/userdata/playlists/video': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/userdata/playlists/music': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/userdata/playlists/mixed': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/userdata/RssFeeds.xml': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/userdata/sources.xml': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/userdata/guisettings.xml': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/userdata/keymaps': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/system': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/media': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/addons/temp': Permission denied
/bin/rm: cannot remove '/data/configuration/miscellanea/kodi/Configuration/addons/packages': Permission denied

Saiyato commented:
I see the problem with the UIConfig, the file, somehow, was name UIconfig (small letter c). Will fix.

The other part is interpretation on my part, I implemented two theories, silly me. On the one hand I thought keeping the config (for reinstallation purposes) would be nice, but I also added a rm -RF in the uninstall script. Since the plugin overwrites existing config files I will delete fashionable. On it!

Saiyato commented:
Created pull #151 for both problems.
